src/main/java/net/sf/jabref/shared/prefs/SharedDatabasePreferences.java

```diff
@@ -89,4 +108,25 @@ public void clear() throws BackingStoreException {
         internalPrefs.clear();
     }
 
+    private Optional<String> getOptionalValue(String key) {
+        return Optional.ofNullable(internalPrefs.get(key, null));
+    }
+
+    public static void clearAll() throws BackingStoreException {
+        Preferences.userNodeForPackage(JabRefMain.class).parent().node(PARENT_NODE).clear();
+    }
+
+    public void putAllDBMSConnectionProperties(DBMSConnectionProperties properties) {
```

Comment:
`DBMSConnectionProperties.isValid()` is performed by methods which open a shared database from the local file. This method is only a shortcut for all `put*` methods.